### PR TITLE
sirius: add rocsolver/wannier90

### PR DESCRIPTION
**Pull request by simonpintarelli**
- add rocsolver dependency for sirius>7.5
- add wannier90 variant (https://github.com/electronic-structure/SIRIUS/pull/838)

### CODE REVIEW

File: var/spack/repos/builtin/packages/sirius/package.py [hunk 1]
```diff
@@ -227,6 +229,9 @@ class Sirius(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("blis threads=openmp", when="+openmp ^blis")
     depends_on("intel-mkl threads=openmp", when="+openmp ^intel-mkl")
 
+    depends_on("wannier90", when="@7.5.0: +wannier90")
+    depends_on("wannier90+shared", when="@7.5.0: +wannier90+shared")
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/sirius/package.py:233]: Is the Wannier90 feature actually going to ship with `7.5.0`?

Comment (simonpintarelli) [reply var/spack/repos/builtin/packages/sirius/package.py:233]: Good point :). I was assuming the PR will get merged before the next release. @toxa81 This is the plan right?

Comment (toxa81) [reply var/spack/repos/builtin/packages/sirius/package.py:233]: 7.5.0 is not tagged yet and not in spack recipe (here); there are a few PRs that need to be merged including the work on Wannier90. To me `when="@7.5.0: +wannier90"` looks fine.

**Review by RMeli — APPROVED** [review]

**Review by toxa81 — APPROVED** [review]
good to be merged

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**